src/vvtbi.cc

```diff
@@ -0,0 +1,729 @@
+                case Tokenizer::TokenType::LT_EQ:
+                    return left <= right;
+                case Tokenizer::TokenType::GT_EQ:
+                    return left >= right;
+                case Tokenizer::TokenType::NOT_EQUAL:
+                    return left != right;
+                default:
+                    dprintf("Internal Error: Invalid comparison operator",
+                            E_ERROR);
+                    return 0;
+            }
+        }
+
+        default:
+            // No comparison operator - treat non-zero as true
+            return left != 0;
+    }
+}
+
+/**
+ * Executes a LET statement.
+ * Format: LET variable = expression
+ *
+ * @throws std::runtime_error on syntax errors
+ */
+void VVTBI::let_statement() {
+    DEBUG_LOG("Processing LET statement");
+
+    // Get variable name
+    auto token = tokenizer_->current_token();
+    if (token != Tokenizer::TokenType::LETTER) {
+        DEBUG_LOG("Expected LETTER, got: " << get_token_string(token));
+        dprintf("Syntax Error: Expected variable name", E_ERROR);
+        return;
+    }
+
+// Process variable
+#ifdef DEBUG_MODE
+    int var_index = tokenizer_->variable_num();
+    char var_name = std::tolower(std::get<char>(tokenizer_->get_token_data()));
+    DEBUG_LOG("Variable name: " << var_name << " (index: " << var_index << ")");
+#else
+    char var_name = std::tolower(std::get<char>(tokenizer_->get_token_data()));
+#endif
+
+    tokenizer_->next_token();
+
+    // Verify and consume equals sign
+    accept(Tokenizer::TokenType::EQUAL);
+
+    // Evaluate the expression
+    DEBUG_LOG("Evaluating expression");
+    int value = expression();
+
+    // Store the value
+    variables_[var_name] = value;
+
+#ifdef DEBUG_MODE
+    DEBUG_LOG("Stored value " << value << " in variable " << var_name
+                              << " at index " << var_index);
+#endif
+}
+/**
+ * Executes an IF statement.
+ * Format: IF condition THEN line_number
+ *
+ * @throws std::runtime_error on syntax errors
+ */
+void VVTBI::if_statement() {
+    DEBUG_LOG("Processing IF statement");
+
+    accept(Tokenizer::TokenType::IF);
+
+    int condition = relation();
+    DEBUG_LOG("Condition result: " << condition);
+
+    accept(Tokenizer::TokenType::THEN);
+
+    if (tokenizer_->current_token() != Tokenizer::TokenType::NUMBER) {
+        dprintf("Syntax Error: Expected line number after THEN", E_ERROR);
+        return;
+    }
+
+    int line_number = tokenizer_->get_num();
+    tokenizer_->next_token();
+
+    if (condition) {
+        DEBUG_LOG("Condition true, jumping to line " << line_number);
+        jump_linenum(line_number);
+        return; // Skip rest of current line
+    }
+
+    // If condition is false, continue to next statement
+    if (tokenizer_->current_token() == Tokenizer::TokenType::EOL) {
+        tokenizer_->next_token();
+    }
+}
+
+/**
+ * Executes a GOTO statement.
+ * Format: GOTO line_number
+ */
+void VVTBI::goto_statement() {
```

Comment:
make sure to write thorough tests for `goto` functions to make sure it works properly